script.js

```diff
@@ -65,8 +76,10 @@ function startWatchLocation() {
     }, error => {
       console.error('WatchPosition error.', error.message, error);
       alert('Failed to find your current location.')
+      reject()
     }, {
```

Comment:
call `reject` on the promise executor when geolocation errors out for more granular control on the result of `startWatchLocation`